src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -54,6 +56,182 @@
+
+            # Get azure-arc agent version for telemetry
+            azure_arc_agent_version = registry_path.split(':')[1]
+
+            # Check for agent verison comaptibility
+            agent_version_check = troubleshootutils.check_agent_version(connected_cluster, azure_arc_agent_version)
+        else:
+            print("Error : Arc Agents have not been added to cluster. Please delete and reconnect the Kubernetes Cluster\n")
+
+        batchv1_api_instance = kube_client.BatchV1Api(kube_client.ApiClient(configuration))
+        utils.validate_node_api_response(batchv1_api_instance, None)
+
+        # Fetching the current kubernetes namespace to deploy the job
+        try:
+            k8s_contexts = config.list_kube_config_contexts()  # returns tuple of (all_contexts, current_context)
+            if kube_context:  # if custom kube-context is specified
+                if k8s_contexts[1].get('name') == kube_context:
+                    current_k8s_context = k8s_contexts[1]
+                else:
+                    for context in k8s_contexts[0]:
+                        if context.get('name') == kube_context:
+                            current_k8s_context = context
+                            break
+            else:
+                current_k8s_context = k8s_contexts[1]
+
+            current_k8s_namespace = current_k8s_context.get('context').get('namespace', "default")  # Take "default" namespace, if not specified in the kube-config
+            namespace_exists = False
+            k8s_v1 = kube_client.CoreV1Api()
+            k8s_ns = k8s_v1.list_namespace()
+            for ns in k8s_ns.items:
+                if ns.metadata.name == current_k8s_namespace:
+                    namespace_exists = True
+                    break
+            if namespace_exists is False:
+                telemetry.set_exception(exception="Namespace doesn't exist", fault_type=consts.Default_Namespace_Does_Not_Exist_Fault_Type,
+                                        summary="The default namespace defined in the kubeconfig doesn't exist on the kubernetes cluster.")
+                raise ValidationError("The default namespace '{}' defined in the kubeconfig doesn't exist on the kubernetes cluster.".format(current_k8s_namespace))
+        except ValidationError as e:
+            raise e
+        except Exception as e:
+            logger.warning("Failed to validate if the active namespace exists on the kubernetes cluster. Exception: {}".format(str(e)))
+
+        # Executing the Diagnoser job check in the given namespace
+        diagnoser_check = troubleshootutils.diagnoser_container_check(corev1_api_instance, batchv1_api_instance, filepath_with_timestamp, current_k8s_namespace)
+
+        # Depending on whether all tests passes we will give the output
+        if (diagnoser_check and msi_cert_check and agent_version_check and arc_agent_state_check and msi_cert_expiry_check and kap_cert_check and kap_security_policy_check):
```

Comment:
we should may be maintain a list of checks and iterate over them

Reply:
Yep i have created a dictionary for that and made all the required changes.